web/src/js/components/contentviews/HttpMessage.tsx

```diff
@@ -60,11 +65,38 @@ export default function HttpMessage({ flow, message }: HttpMessageProps) {
         }
     }, [content]);
 
-    if (edit) {
+    const handleClickCopyButton = () => {
+        const url = MessageUtils.getContentURL(flow, message, contentView); //without the 'maxLines' parameter, so we can get the full content of the content view
+
+        setIsFetchingFullContent(true);
+
+        fetchApi(url)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error(
+                        `${response.status} ${response.statusText}`.trim(),
+                    );
+                }
+                return response.json();
+            })
+            .then((data: ContentViewData) => {
+                copyViewContentDataToClipboard(data);
+                setIsCopied(true);
+                setTimeout(() => setIsCopied(false), 2000);
+            })
+            .catch((e) => {
+                console.error(e);
```

Comment:
This PR seems like a great place for me to learn about mitmweb :smile: . Sorry for delaying this PR.

I wonder if there is currently any error handling in similar places? In this case something like an error toast would be great. Or are there other places were we also simply `console.error`? The user should probably get some sort of feedback right?

Reply:
It depends, actually. In some cases, we display error messages as red text (e.g., on the capture page and in the options modal). However, in other cases like this one, displaying an error message is tricky because, to be honest, I’m not sure where it would fit best for an optimal UX. Using toasts could be a good alternative, and I think we should discuss whether we want to adopt toasts consistently across mitmweb. For now, to keep things simple, I’m just logging the error to the console.